### PR TITLE
Fix test failures on Windows OneCoreUAP

### PR DESCRIPTION
**Pull request by JohnChen0**
This PR fixes all test failures that I encountered while running the inner-loop tests on Windows OneCoreUAP SKU.

### CODE REVIEW

**Comment by gkhanna79** [comment]
LGTM.

@weshaggard @stephentoub PTAL as well.

File: src/Common/tests/System/PlatformDetection.cs [hunk 1]
```diff
@@ -23,6 +23,9 @@ public static class PlatformDetection
             File.Exists(Path.Combine(Environment.GetEnvironmentVariable("windir"), "regedit.exe")));
```
Comment (weshaggard) [inline src/Common/tests/System/PlatformDetection.cs]: Why I know this isn't your change I'm not sure why we chose regedit for a Nano feature query. @Priya91 any reason we chose that program to key off of?

**Comment by weshaggard** [comment]
cc @davidsh @CIPop  for any feedback on the httpapi query.

File: src/Common/tests/System/PlatformDetection.cs (hunk 1 above)
Comment (Priya91) [inline src/Common/tests/System/PlatformDetection.cs]: Nano doesn't support any GUI programs, so used regedit as every other windows platform had with the previous requirements.

File: src/Common/tests/System/PlatformDetection.cs [hunk 2]
```diff
@@ -23,6 +23,9 @@ public static class PlatformDetection
...
         public static bool IsWindows10Version1607OrGreater { get; } = IsWindows &&
             GetWindowsVersion() == 10 && GetWindowsMinorVersion() == 0 && GetWindowsBuildNumber() >= 14393;
+        // Windows OneCoreUAP SKU doesn't have httpapi.dll
+        public static bool HasHttpApi { get; } = (IsWindows &&
```
Comment (CIPop) [inline src/Common/tests/System/PlatformDetection.cs]: If on UAP all http.sys/httpapi.dll tests should be disabled (temporarily until we get to implement it properly on UAP).
The check doesn't hold when running these tests on a full Windows machine in UAP mode.

**Comment by CIPop** [comment]
Thanks @weshaggard. @tarekgh While the check works fine on OneCoreUAP, it will break once UAP tests run on dev boxes running Windows client.

**Comment by tarekgh** [comment]
@CIPop, I am not really the owner of the test or did the logic. please go ahead and fix the test or wait @JohnChen0 till he is back. sorry I don't have cycles to look at this now.

**Comment by gkhanna79** [comment]
@CIPop Why will the check break on Windows client and which specific version?

**Comment by CIPop** [comment]
> @CIPop, I am not really the owner of the test or did the logic. please go ahead and fix the test or wait @JohnChen0 till he is back. sorry I don't have cycles to look at this now.

No worries: I didn't even know we're running UAP testing until today. The implementation will change at some point anyway and we'll remove the check.

> Why will the check break on Windows client and which specific version?

Windows Desktop has the DLL yet the APIs are not accessible in UWP context.

File: src/System.Diagnostics.Process/tests/ProcessTests.cs [hunk 3]
```diff
@@ -595,8 +595,13 @@ private static bool ProcessPeformanceCounterEnabled()
         {
```
Comment (stephentoub) [inline src/System.Diagnostics.Process/tests/ProcessTests.cs]: Nit: Pre-existing your change, but Peformance should be Performance

File: src/System.Net.HttpListener/tests/AuthenticationTests.cs [hunk 4]
```diff
@@ -17,7 +17,7 @@ public class AuthenticationTests
         private const string TestUser = "testuser";
         private const string TestPassword = "testpassword";
 
-        [Theory]
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.HasHttpApi))]
```
Comment (stephentoub) [inline src/System.Net.HttpListener/tests/AuthenticationTests.cs]: @Priya91, @JohnChen0, I still see this HasHttpApi checking IsWindows... doesn't that mean that even though the PlatformSpecific was removed from the test class, none of these tests are actually running on Unix now?

Comment (Priya91) [inline src/System.Net.HttpListener/tests/AuthenticationTests.cs]: Yeah.. These are checking for httpapi.dll to be present on box. Will fix this.

**Pull request merged**